### PR TITLE
Emit confirmed info event per confirmed block

### PR DESCRIPTION
**Pull request by gitferry**
This PR introduce the emitting of confirmed info event before finishing handling a confirmed block. This sends the confirmed height along with confirmed tvl in the event. Given that the indexer ensures all the confirmed blocks are handled in the order of block height, no tvl data will be missed

### CODE REVIEW

File: indexer/indexer.go [hunk 1]
```diff
@@ -383,6 +383,16 @@ func (si *StakingIndexer) HandleConfirmedBlock(b *types.IndexedBlock) error {
 		return fmt.Errorf("failed to save the last processed height: %w", err)
 	}
 
+	// emit ConfirmedInfoEvent to send the confirmed height and tvl
+	confirmedTvl, err := si.is.GetConfirmedTvl()
```
Comment (KonradStaniec) [inline indexer/indexer.go:387]: wdyt about having:
1. `confirmedTvlBeforeProcessing`
2. `confirmedTvlAfterProcessing`
3. Pushing the event only if tvl have changed ? This should reduce spam a bit, as for empty blocks (without staking transaction)there will be not events. Drawback is that first event will be pushed only with the first confirmed staking transaction

**Review by KonradStaniec — APPROVED** [review]

File: indexer/indexer.go (hunk 1 above)
Comment (gitferry) [reply indexer/indexer.go:387]: I think maybe it's easier for the api to expect an event per block, no? @jrwbabylonchain

File: indexer/indexer.go [hunk 2]
```diff
@@ -383,6 +383,16 @@ func (si *StakingIndexer) HandleConfirmedBlock(b *types.IndexedBlock) error {
...
+	if err != nil {
+		return fmt.Errorf("failed to get the confirmed tvl: %w", err)
+	}
+	confirmedInfoEvent := queuecli.NewConfirmedInfoEvent(uint64(b.Height), confirmedTvl)
+	if err := si.consumer.PushConfirmedInfoEvent(&confirmedInfoEvent); err != nil {
```
[the remark this answers is not in the document]
Comment (gitferry) [reply indexer/indexer.go]: Good point. Added it in [ac9a415](https://github.com/babylonchain/staking-indexer/pull/143/commits/ac9a41555a9aa3d85d64a3fcdb5cbc0fed5aae27)

File: config/config.go [hunk 3]
```diff
@@ -31,12 +31,13 @@ var (
...
 type Config struct {
-	LogLevel       string         `long:"loglevel" description:"Logging level for all subsystems" choice:"trace" choice:"debug" choice:"info" choice:"warn" choice:"error" choice:"fatal"`
-	BitcoinNetwork string         `long:"bitcoinnetwork" description:"Bitcoin network to run on" choice:"mainnet" choice:"regtest" choice:"testnet" choice:"simnet" choice:"signet"`
-	BTCConfig      *BTCConfig     `group:"btcconfig" namespace:"btcconfig"`
-	DatabaseConfig *DBConfig      `group:"dbconfig" namespace:"dbconfig"`
-	QueueConfig    *QueueConfig   `group:"queueconfig" namespace:"queueconfig"`
-	MetricsConfig  *MetricsConfig `group:"metricsconfig" namespace:"metricsconfig"`
+	LogLevel          string         `long:"loglevel" description:"Logging level for all subsystems" choice:"trace" choice:"debug" choice:"info" choice:"warn" choice:"error" choice:"fatal"`
+	BitcoinNetwork    string         `long:"bitcoinnetwork" description:"Bitcoin network to run on" choice:"mainnet" choice:"regtest" choice:"testnet" choice:"simnet" choice:"signet"`
+	ExtraEventEnabled bool           `long:"extraeventenabled" description:"Whether emitting non-default events is allowed"`
```
[the remark this answers is not in the document]
Comment (gitferry) [reply config/config.go:36]: per offline discussion, we decided to keep using bool for now as implementing enum is overkill for only one option. We can revisit this when more extra events are introduced

**Pull request merged**